File: test-suite-python/src/test/python/micronaut/servlet/docs/tomcat/TomcatServerCustomizer.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener
from org.apache.catalina.startup import Tomcat
# end::imports[]
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="TomcatServerCustomizerTest")
# tag::class[]
@Singleton
class TomcatServerCustomizer(BeanCreatedEventListener[Tomcat]):

    def onCreated(self, event: BeanCreatedEvent[Tomcat]) -> Tomcat:
        tomcat = event.getBean()
        # perform customizations, for example:
        tomcat.getServer().setUtilityThreads(2)
        return tomcat
# end::class[]

File: test-suite/src/test/java/io/micronaut/servlet/docs/tomcat/TomcatServerCustomizer.java
package io.micronaut.servlet.docs.tomcat;

// tag::imports[]
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import org.apache.catalina.startup.Tomcat;
import jakarta.inject.Singleton;
// end::imports[]
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "TomcatServerCustomizerTest")
// tag::class[]
@Singleton
public class TomcatServerCustomizer implements BeanCreatedEventListener<Tomcat> {
    @Override
    public Tomcat onCreated(BeanCreatedEvent<Tomcat> event) {
        Tomcat tomcat = event.getBean();
        // perform customizations, for example:
        tomcat.getServer().setUtilityThreads(2);
        return tomcat;
    }
}
// end::class[]
